src/time.js

```diff
@@ -41,14 +42,22 @@ const utcIntervals = new Map([
   ["sunday", utcSunday]
 ]);
 
+function isoInterval(interval) {
+  return Object.assign((t) => interval(t), {...interval, floor: (t) => interval.floor(coerceDate(t))});
```

Comment:
This feels awkward to me. Like, isTimeInterval is the way it is because that was a quick and dirty way to test for a D3 time interval. But now we have to recreate that representation even though we don’t need it to satisfy the isTimeInterval check. And this is only adding coercion for _one_ of the possible methods (floor) on the time interval, not the main function, not _interval_.offset and _interval_.range, etc. I’m not sure what fix I recommend yet.

Reply:
We can give them the same treatment, but in practice don't use them on user-provided data. Their input is in all cases already coerced.

(Agree it's awkward!)